staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -109,9 +109,14 @@ func InstrumentRouteFunc(verb, resource string, routeFunc restful.RouteFunction)
 }
 
 func cleanUserAgent(ua string) string {
+	// We collapse all "web browser"-type user agents into one "browser" to reduce metric cardinality.
 	if strings.HasPrefix(ua, "Mozilla/") {
 		return "Browser"
 	}
+	// If an old "kubectl.exe" has passed us its full path, we discard the path portion.
+	if exeIdx := strings.LastIndex(strings.ToLower(ua), "kubectl.exe"); exeIdx != -1 {
+		return ua[exeIdx:]
```

Comment:
This can panic based on user input since `strings.ToLower(ua)` is not necessarily the same length in bytes as `ua`:

https://play.golang.org/p/0BV2H6OlKi